### PR TITLE
checker(dm): move checker from tidb-tools to dm

### PR DESCRIPTION
**Pull request by okJiang**
### What problem does this PR solve? 

close https://github.com/pingcap/ticdc/issues/3768

### What is changed and how it works?

- Move checker from [tidb-tools](https://github.com/pingcap/tidb-tools/tree/master/pkg/check) to dm

### Check List 

Tests 

 - Unit test
 - Integration test

### Release note 

```release-note
None
```

### CODE REVIEW

File: dm/checker/mysql_server.go [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+	Min MySQLVersion
+	Max MySQLVersion
+}{
+	"mysql": {
+		MySQLVersion{5, 6, 0},
+		MaxVersion,
+	},
+	"mariadb": {
+		MySQLVersion{10, 1, 2},
+		MaxVersion,
```
Comment (lance6716) [inline dm/checker/mysql_server.go:49]: remember that newer MariaDB doesn't fit DM well. maybe we should be more strict about the version and this check could raise warning not error? (in next PR)

https://github.com/pingcap/dm/issues/2266

**Review by lance6716 — COMMENTED** [review]
could you explain what's the modification you made to get this PR? it's a bit long to review 😵

**Comment by okJiang** [comment]
> could you explain what's the modification you made to get this PR? it's a bit long to review 😵

no modification for the checker. just move it. @lance6716

File: dm/checker/mysql_server.go (hunk 1 above)
Comment (okJiang) [reply dm/checker/mysql_server.go:49]: MariaDB seems to have a very different permission limit than MySQL, I wonder if we can not report MariaDB error.

Or we could just not support MariaDB in pre-check until we sorted out a better MariaDB support scheme.

File: dm/checker/binlog.go [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package checker
```
Comment (Ehco1996) [inline dm/checker/binlog.go:14]: maybe move this package to  `ticdc/pkg/checker`  or `ticdc/dm/pkg/` ? https://github.com/golang-standards/project-layout#pkg

**Review by Ehco1996 — COMMENTED** [review]
this too large too review 😂 but if you just copy code from `tidb/tidb-tools` i think this pr is ready to merge becase CI passed

File: dm/checker/binlog.go (hunk 2 above)
Comment (okJiang) [reply dm/checker/binlog.go]: done in https://github.com/pingcap/ticdc/pull/3770/commits/7b672e89a4116ee2adfbe68645084a69ae091c5c

**Comment by okJiang** [comment]
The CI passed all.  ptal again❤️ @Ehco1996 @lance6716

**Review by Ehco1996 — APPROVED** [review]

**Review by GMHDBJD — APPROVED** [review]
LGTM

**Pull request merged**